clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security/AmazonCredentials.java

```diff
@@ -125,6 +134,178 @@ public String getDefaultKeyPair() {
         return regions;
     }
 
+    public AWSProxy getAWSProxy() {
+        return proxy;
+    }
+
+
+    @ConfigurationProperties(prefix = "aws.proxy", locations = "clouddriver.yml")
+    @Component
+    public static class AWSProxy {
```

Comment:
You should be able to just make this a Groovy @Canonical class:
````groovy
class AWSProxy {
  String proxyHost
  String proxyPort
  String proxyUsername
  String proxyPassword
  String proxyDomain
  String proxyWorkstation
  String protocol
}
````
and add your extra business logic methods (`apply`, `isNTLMProxy`, `isProxyMode`)